filters/openpolicyagent/internal/envoy/envoyplugin.go

```diff
@@ -0,0 +1,121 @@
+package envoy
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/plugins"
+	"github.com/open-policy-agent/opa/util"
+)
+
+// Factory defines the interface OPA uses to instantiate a plugin.
+type Factory struct{}
+
+// New returns the object initialized with a valid plugin configuration.
+func (Factory) New(m *plugins.Manager, cfg interface{}) plugins.Plugin {
+	p := &Plugin{
+		manager: m,
+		cfg:     *cfg.(*PluginConfig),
+	}
+
+	m.UpdatePluginStatus(PluginName, &plugins.Status{State: plugins.StateNotReady})
+
+	return p
+}
+
+// Validate returns a valid configuration to instantiate the plugin.
+func (Factory) Validate(m *plugins.Manager, bs []byte) (interface{}, error) {
+	cfg := PluginConfig{
+		DryRun: defaultDryRun,
+	}
+
+	if err := util.Unmarshal(bs, &cfg); err != nil {
+		return nil, err
+	}
+
+	if err := cfg.ParseQuery(); err != nil {
+		return nil, err
+	}
+
+	return &cfg, nil
+}
+
+func (p *Plugin) Reconfigure(ctx context.Context, config interface{}) {
+
```

Comment:
missing or not needed? Maybe doc why it's not implemented.

Reply:
Part of the interface. Will document why the empty implementation is ok.

Reply:
This funnily enough is actually a bug that also exists in the OPA Envoy plugin. It should actually never materialise but it would still be surprising behaviour